### ISSUE TITLE
GroupNorm is slow stragely when calling backward() in pytorch 1.6.0

### ISSUE THREAD

**Issue by SpartaG117** [pytorch/pytorch#46086]
## 🐛 Bug

Calling loss.backward() when using GroupNorm layer becomes very slow in Pytorch 1.6.0.  
It is almost two times slower than it in 1.5.1

## To Reproduce

    import torch
    import time
    import torch.nn as nn
    norm1 = nn.GroupNorm(8, 512).cuda()
    norm2 = nn.GroupNorm(8, 512).cuda()
    s = time.time()
    for i in range(1000):
        a = torch.rand((1024, 512)).float().cuda().requires_grad_()
        b = torch.rand((1024, 512)).float().cuda().requires_grad_()
        ag = norm1(a)
        bg = norm2(b)
        loss = (ag + bg).reshape(-1).mean()
        loss.backward()
    print(time.time()-s)

When using pytorch 1.5.1:

    10.412362575531006

When using pytorch 1.6.0:

    20.195667266845703

When commenting loss.backward(), it derives the similar number. But 1.6.0 is still a bit slower than 1.5.1.

1.5.1

    9.152965545654297
1.6.0

    10.689554214477539
## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.6.0 and 1.5.1
 - OS (e.g., Linux):  Ubuntu 16.04.
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: 10.1.243
 - GPU models and configuration: Tesla P40
 - Any other relevant information:

## Additional context

cc @albanD @mruberry @VitalyFedyunin @ngimel

**Comment by ngimel** [comment]
cc @BIT-silence

**Comment by BIT-silence** [comment]
I just take a look at this case. I tested it on a V100 machine and the gap is much smaller, but it is still slower than v1.5.1.  I think the reason the feature_size of this test case is 512,  then each group only has 512/8 = 64 features, which is pretty small. Our optimization in https://github.com/pytorch/pytorch/pull/28204 is mainly for CV use cases, where consider C * H * W will be much larger. 

So in this case, if our reduced_size is 64, the impl will have some issues which didn't fully use the num of cuda threads. I will prepare one PR to improve the performance in the cases that are not in CV context.

**Comment by BIT-silence** [comment]
https://github.com/pytorch/pytorch/pull/46170 should fix this.

**Comment by SpartaG117** [comment]
@BIT-silence Thank you！

**Comment by ngimel** [comment]
Closing, please reopen if the issue persists.
